sdks/java/io/hadoop-format/src/main/java/org/apache/beam/sdk/io/hadoop/format/HadoopFormatIO.java

```diff
@@ -475,6 +479,18 @@
       return withValueTranslation(function).toBuilder().setValueCoder(coder).build();
     }
 
+    /**
+     * Determines if key-value clone should be skipped or not (default is 'false'). Hadoop formats
+     * typically work with Writable data structures which are mutable. Therefore, this IO will clone
+     * read key-values if they are not in the list of well known immutable types. However, in case
+     * user does use key/value translation functions, resulting key-values might already be
```

Comment:
Should it be used only with `withKeyTranslation()` or `withValueTranslation()`, or it can be used independently?

Reply:
I would keep it separate. I am not sure if there are some input formats out there which can be not reusing instances of does not use writables for serialization.